packages/help-center/src/components/help-center-content.tsx

```diff
@@ -121,10 +139,9 @@ const HelpCenterContent: React.FC< { isRelative?: boolean; currentRoute?: string
 		}
 	}, [ navigate, navigateToRoute, setNavigateToRoute, location ] );
 
-	// reset the scroll location on navigation, TODO: unless there's an anchor
 	useEffect( () => {
 		setSearchTerm( '' );
-		if ( containerRef.current ) {
+		if ( containerRef.current && ! location.hash && ! location.pathname.includes( '/odie' ) ) {
```

Comment:
If I understood well, this change together with the better place of the <ProtectedRoute /> should resolve the issue.

Reply:
Well, this is more preventive than any other thing, I don't want scroll being handled outside of the Wapuu client, as it is itself handling scroll behavior. I also saw that there were a TODO, I've seen it many times while I was doing some other side developments and it was driving me mad seeing that TODO all the time there 😆

Reply:
Thanks, I have this [PR](https://github.com/Automattic/wp-calypso/pull/93624/files) to address another issue that touches this part of the code, I will revisit this PR and make sure it won't impact `/odie`